File: SDKs/googleapis.com/cloudtasks/v2beta2/python/src/sdk/models/shared/taskstatus.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import attemptstatus as shared_attemptstatus


@dataclass_json
@dataclasses.dataclass
class TaskStatus:
    r"""TaskStatus
    Status of the task.
    """
    
    attempt_dispatch_count: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('attemptDispatchCount') }})
    attempt_response_count: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('attemptResponseCount') }})
    first_attempt_status: Optional[shared_attemptstatus.AttemptStatus] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('firstAttemptStatus') }})
    last_attempt_status: Optional[shared_attemptstatus.AttemptStatus] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('lastAttemptStatus') }})
    

File: SDKs/googleapis.com/cloudtasks/v2beta2/java/lib/src/main/java/openapisdk/models/shared/TaskStatus.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * TaskStatus
 * Status of the task.
**/
public class TaskStatus {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("attemptDispatchCount")
    public Integer attemptDispatchCount;
    public TaskStatus withAttemptDispatchCount(Integer attemptDispatchCount) {
        this.attemptDispatchCount = attemptDispatchCount;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("attemptResponseCount")
    public Integer attemptResponseCount;
    public TaskStatus withAttemptResponseCount(Integer attemptResponseCount) {
        this.attemptResponseCount = attemptResponseCount;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("firstAttemptStatus")
    public AttemptStatus firstAttemptStatus;
    public TaskStatus withFirstAttemptStatus(AttemptStatus firstAttemptStatus) {
        this.firstAttemptStatus = firstAttemptStatus;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("lastAttemptStatus")
    public AttemptStatus lastAttemptStatus;
    public TaskStatus withLastAttemptStatus(AttemptStatus lastAttemptStatus) {
        this.lastAttemptStatus = lastAttemptStatus;
        return this;
    }
}